README.txt

Note: This is dev verison of the module. Please test and review the module and post
your issues on the project page http://drupal.org/project/ddblock to help improve
the module.


SUMMARY
-------
The Dynamic display block module enables you to create blocks to present content
in a dynamic way. For creating slideshow effects it uses the jQuery Cycle plug-in.

There are four ways to specify content for the Dynamic display block module:

* A folder containing image files
* A node from a content type with multiple input
* Views
* An instance of any created block

Several effects and other setting can be set in the configuration settings of the
Dynamic display block module.


MORE INFORMATION
----------------
Documentation in progress


REQUIREMENTS
------------
None


INSTALLATION
------------
See INSTALL.txt


COLLABORATION WITH OTHER MODULES
--------------------------------
The dynamic display block module can work together with all modules that provide
blocks. The content of the block can be displayed dynamically (one by one width
transition effects) if the content consist of e.g. Images, A list of listitems,
A table with tablerows, etc.

Example modules and blocks you can use

Module       - blocks
---------------------
Headerimage  - blocks with images
Comments     - Recent comments
Forum        - Active forum topics
Forum        - New forum topics
Views        - blocks with images, lists, tables
Worldclock   - World clock
etc.

* Set the blocks you want to use with the dynamic display block module in
  Administer >> Site configuration >> Dynamic display block - Settings tab.

* Make an instance of the block you want to use in Administer >> Site
  configuration >> Dynamic display block - Instance tab.

* Configure the block in Administer >> Site configuration >> Dynamic display
  block - List tab.

  Dont forget to set the Container value to the tag you want to show

* Add the block to a region in in Administer >> Site Building >> Blocks.

CONFIGURATION
-------------
* Configure user permissions in the modules page: Modules

  - administer dynamic display blocks:

  - view dynamic display blocks:

* Manage module settings in Structure >> Dynamic display block.


ADVANCED SLIDESHOWS
-------------------
Documentation in progress

FAQ
---
For Frequently Asked Questions - http://ddblock.myalbums.biz/faq

CONTACT
-------
Original idea cycle module by roopletheme - http://drupal.org/user/164625

Major rewrite by current Maintainers:
* Philip Blaauw (ppblaauw) - http://drupal.org/user/155138

We also offer installation, development, theming, customization.
You can contact us via the contact form on http://ddblock.myalbums.biz.
or via email to ppblaauw (at) gmail.com

CHANGELOG.txt
/*
 * Change Log for ddblock module
 */

2011-JUL-23 
#1226554 fix pause on hover always used in basic options

2011-JUL-18
Add easing support for slide and slidetext using jquery UI
#1190082 improve help text for choosing content types
add nr_of_items functionality for custom pager
adjust nr_of_items help text
improve help text for before and after effects
Cycle plugin now via libraries


2011-FEB-14
#1058868 Adjust install.txt
bug-fix  Add pagerPosition to javascript settings for horizontal/vertical scrollable pager
bug-fix  Add pager_toggle to variables given to content template file to toggle the pager on/off
bug-fix  add hyphen (-) to mask for finding size-xxx files

2011-FEB-11
#1049556 fix delete of blocks
#1053996 fix warnings when no images
#1055048 fix remove extra slide div not closed
#1054488 fix remove extra commit button on config page
 
2011-JAN-29 
Fixed redirects after forms are submitted
Rename function _dir_scan_directory to _ddblock_dir_scan_directory
Fixed Notice: Undefined property: stdClass::$ddblock_enabled on delete form 
Fixed db_affected_rows() does not exist anymore in Drupal 7
Cleanup CHANGELOG.TXT file

2011-JAN-25 
Initial version



ddblock.class.php
<?php
class ddblockConfigurationSettings {

  /**
   * @var array default configuration settings
   */
  protected $settings = array(
    'content_type'             => array('format' => '%s', 'value' => 'none'),
    'folder'                   => array('format' => '%s', 'value' => 'images/ddblock'),
    'ignore_files'             => array('format' => '%s', 'value' => ''),
    'input_type'               => array('format' => '%s', 'value' => 'images'),
    'nodes'                    => array('format' => '%s', 'value' => ''),
    'node_body_teaser'         => array('format' => '%s', 'value' => 'body'),
    'order'                    => array('format' => '%s', 'value' => 'asc'),
    'output'                   => array('format' => '%s', 'value' => 'view_content'),
    'widget'                   => array('format' => '%s', 'value' => 'cycle'),
  );

  public function __construct (Array $settings) {
    foreach ($settings as $key => $value) {
      $this->__set($key, $value);
    }
  }

  /**
   * Use overload functions to get and set $settings
   *
   * Keys passed to the set function should be keys that already
   * exist in the settings array.
   */
  public function __set($key, $value) {
    if (!array_key_exists($key, $this->settings)) {
      return;
    }
    $this->settings[$key]['value'] = sprintf($this->settings[$key]['format'], $value);
  }

  /**
   * Use overload functions to get and set $settings
   */
  public function __get($key) {
    if (!array_key_exists($key, $this->settings)) {
      return null;
    }
    return $this->settings[$key]['value'];
  }

  /**
   * Alternate function to __get
   */
  public function get($key, $default) {
    return ($this->__get($key) !== NULL) ? $this->__get($key) : $default;
  }

  public function settings() {
    $settings = array();
    foreach ($this->settings as $key => $value) {
      $settings[$key] = $value['value'];
    }
    return $settings;
  }
}

class ddblockCycleConfigurationSettings {

  /**
   * @var array default configuration settings
   */
  protected $settings = array(
    'advanced'                 => array('format' => '%d', 'value' => 1),
    'container_advanced'       => array('format' => '%s', 'value' => 'div.slide'),
    'container_basic'          => array('format' => '%s', 'value' => 'div.slide'),
    'custom_jquery'            => array('format' => '%s', 'value' => ''),
    'fx'                       => array('format' => '%s', 'value' => 'fade'),
    'easing_out'               => array('format' => '%s', 'value' => 'linear'),
    'easing_in'                => array('format' => '%s', 'value' => 'linear'),
    'height'                   => array('format' => '%d', 'value' => 195),
    'image_height'             => array('format' => '%d', 'value' => 183),
    'image_width'              => array('format' => '%d', 'value' => 183),
    'image_style_toggle'       => array('format' => '%d', 'value' => 0),
    'image_style_slide'        => array('format' => '%s', 'value' => '<none>'),
    'image_style_pager_item'   => array('format' => '%s', 'value' => '<none>'),
    'max_image'                => array('format' => '%d', 'value' => 5),
    'next'                     => array('format' => '%d', 'value' => 0),
    'nr_of_pager_items'        => array('format' => '%d', 'value' => 4),
    'overflow'                 => array('format' => '%d', 'value' => 1),
    'pager'                    => array('format' => '%s', 'value' => 'none'),
    'pager_container'          => array('format' => '%s', 'value' => 'pager-item'),
    'pager_disable_click'      => array('format' => '%d', 'value' => 1),
    'pager_event'              => array('format' => '%s', 'value' => 'click'),
    'pager_fast'               => array('format' => '%d', 'value' => 1),
    'pager_height'             => array('format' => '%d', 'value' => 25),
    'pager_pause'              => array('format' => '%d', 'value' => 1),
    'pager_width'              => array('format' => '%d', 'value' => 195),
    'pager_position'           => array('format' => '%s', 'value' => 'top'),
    'pager_prev_next_loop'     => array('format' => '%d', 'value' => 1),
    'pager_scrollable_loop'    => array('format' => '%d', 'value' => 1),
    'pager_toggle'             => array('format' => '%d', 'value' => 1),
    'pager2'                   => array('format' => '%d', 'value' => 0),
    'pager2_event'             => array('format' => '%s', 'value' => 'click'),
    'pager2_slide_prev'        => array('format' => '%s', 'value' => ''),
    'pager2_slide_next'        => array('format' => '%s', 'value' => ''),
    'pager2_slide_hide'        => array('format' => '%d', 'value' => 0),
    'pause'                    => array('format' => '%d', 'value' => 1),
    'slide_text'               => array('format' => '%d', 'value' => 1),
    'slide_text_after_effect'  => array('format' => '%s', 'value' => 'fadeIn'),
    'slide_text_after_speed'   => array('format' => '%d', 'value' => 1000),
    'slide_text_after_easing'  => array('format' => '%s', 'value' => 'linear'),
    'slide_text_before_effect' => array('format' => '%s', 'value' => 'fadeOut'),
    'slide_text_before_speed'  => array('format' => '%d', 'value' => 250),
    'slide_text_before_easing' => array('format' => '%s', 'value' => 'linear'),
    'slide_text_container'     => array('format' => '%s', 'value' => 'div.slide-text'),
    'slide_text_jquery'        => array('format' => '%d', 'value' => 0),
    'slide_text_position'      => array('format' => '%s', 'value' => 'bottom'),
    'speed'                    => array('format' => '%d', 'value' => 500),
    'template'                 => array('format' => '%s', 'value' => 'none'),
    'template_size'            => array('format' => '%s', 'value' => 'default'),
    'timeout'                  => array('format' => '%d', 'value' => 5000),
    'width'                    => array('format' => '%d', 'value' => 195),
    // isis modified
    'use_cross_slide'          => array('format' => '%d', 'value' => 0),
    'cover_image_url'          => array('format' => '%s', 'value' => ''),
  );

  public function __construct (Array $settings) {
    foreach ($settings as $key => $value) {
      $this->__set($key, $value);
    }
  }

  /**
   * Use overload functions to get and set $settings
   *
   * Keys passed to the set function should be keys that already
   * exist in the settings array.
   */
  public function __set($key, $value) {
    if (!array_key_exists($key, $this->settings)) {
      return;
    }
    $this->settings[$key]['value'] = sprintf($this->settings[$key]['format'], $value);
  }

  /**
   * Use overload functions to get and set $settings
   */
  public function __get($key) {
    if (!array_key_exists($key, $this->settings)) {
      return null;
    }
    return $this->settings[$key]['value'];
  }

  /**
   * Alternate function to __get
   */
  public function get($key, $default) {
    return ($this->__get($key) !== NULL) ? $this->__get($key) : $default;
  }

  public function settings() {
    $settings = array();
    foreach ($this->settings as $key => $value) {
      $settings[$key] = $value['value'];
    }
    return $settings;
  }
}

class ddblockCycleMappingSettings {

  /**
   * @var array default configuration settings
   */
  protected $settings = array(
    0  => array('target' => 'node_id', 'source' => '', ),
    1  => array('target' => 'slide_image', 'source' => '', ),
    2  => array('target' => 'slide_title', 'source' => '', ),
    3  => array('target' => 'slide_text', 'source' => '', ),
    4  => array('target' => 'slide_read_more', 'source' => '', ),
    5  => array('target' => 'pager_image', 'source' => '', ),
    6  => array('target' => 'pager_text', 'source' => '', ),
    7  => array('target' => '', 'source' => '', ),
    8  => array('target' => '', 'source' => '', ),
    9  => array('target' => '', 'source' => '', ),
    10 => array('target' => '', 'source' => '', ),
    11 => array('target' => '', 'source' => '', ),
    12 => array('target' => '', 'source' => '', ),
    13 => array('target' => '', 'source' => '', ),
    14 => array('target' => '', 'source' => '', ),
    15 => array('target' => '', 'source' => '', ),
  );

  public function __construct (Array $settings) {
    foreach ($settings as $key => $value) {
      $this->__set($key, $value);
    }
  }

  /**
   * Use overload functions to get and set $settings
   *
   * Keys passed to the set function should be keys that already
   * exist in the settings array.
   */
  public function __set($key, $value) {
    if (!array_key_exists($key, $this->settings)) {
      return;
    }
    $this->settings[$key]['target'] = $value['target'];
    $this->settings[$key]['source'] = $value['source'];
  }

  /**
   * Use overload functions to get and set $settings
   */
  public function __get($key) {
    if (!array_key_exists($key, $this->settings)) {
      return null;
    }
    return $this->settings[$key];
  }

  /**
   * Alternate function to __get
   */
  public function get($key, $default) {
    return ($this->__get($key) !== NULL) ? $this->__get($key) : $default;
  }

  public function settings() {
    $settings = array();
    foreach ($this->settings as $key => $value) {
      $settings[$key]['target'] = $value['target'];
      $settings[$key]['source'] = $value['source'];
    }
    return $settings;
  }
}
